airbyte-integrations/connectors/source-asana/acceptance-test-config.yml

```diff
@@ -1,21 +1,33 @@
 # See [Source Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference)
 # for more information about how to configure these tests
 connector_image: airbyte/source-asana:dev
-tests:
+test_strictness_level: high
+acceptance_tests:
   spec:
-    - spec_path: "source_asana/spec.json"
+    tests:
+      - spec_path: "source_asana/spec.json"
   connection:
-    - config_path: "secrets/config.json"
-      status: "succeed"
-    - config_path: "integration_tests/invalid_config.json"
-      status: "failed"
+    tests:
+      - config_path: "secrets/config.json"
+        status: "succeed"
+      - config_path: "integration_tests/invalid_config.json"
+        status: "failed"
   discovery:
-    - config_path: "secrets/config.json"
+    tests:
+      - config_path: "secrets/config.json"
   basic_read:
-    - config_path: "secrets/config.json"
-      configured_catalog_path: "integration_tests/configured_catalog.json"
-      timeout_seconds: 7200
+    tests:
+      - config_path: "secrets/config.json"
+        timeout_seconds: 7200
+        expect_records:
+          path: "integration_tests/expected_records.txt"
+        empty_streams:
+          - name: custom_fields
+            bypass_reason: "This stream in not available for our Test Account. 402 - Custom Fields are not available for free users or guests."
```

Comment:
Can we create a dedicated paid sandbox account to cover this stream?

Reply:
cc @YowanR to make decisions here)

Reply:
It seems that the main issue that we are running into is that we are using an internal account instead of a sandbox account. For most connectors, we should have sandbox accounts instead of prod account. Now, a big issue here is that we actually took a very long time to get this account unblocked for OAuth. If we were to make another account, I would expect that we would block for a long time again which is detrimental to our customers. Instead, I suggest that we do two things: 
1) Bypass the tests that are failing due to having a prod account and not a sandbox acc (plus hide sensitive info!) 
2) Start setting up a sandbox account to revisit. Once we have the sandbox account, we can solve the issues that we've seen. Thoughts on this @airbytehq/connector-operations and @bazarnov ?

Reply:
@YowanR it's unclear to me if you want this PR to certify Asana to GA.
If it's not, I'm fine with logging future work for creating a sandbox account and adding the bypass reasons. I don't want to block any feature improvement on the connector. But I'd love new GA connectors to have the testing guarantees we outlined.

Reply:
@alafanechere I'd like us to use strict testing to get Asana to GA but, because of how hard it was to get OAuth for this account, we need to work with the restrictions of using a prod account instead of a sandbox account. Based on the comments you've left in this PR, it seems that we will need to bypass a few tests, but overall, we'll still covering most tests correct? I think that the main question I have is: do you think the strict SAT framework can work with a prod account or is that a no-no?

Reply:
The use of an internal account requires us to bypass a subset of basic reads test (expected records) and full-refresh as @bazarnov mentioned flaky tests results due to interaction with our internal account.

I think I don't get the relationship between GA and OAuth. 
Can't we keep a `beta` release stage on Asana with OAuth (with `test_strictness_level` set to `high` + a couple of tests to bypass)? And then certify to GA when we have the sandbox account?  Or does not certifying Asana to GA puts us in a difficult situation customer-wise?

Reply:
Adding my 2 cents - I think that GA means that we can reliably test the connector and guarantee that everything works, and continues to test all the relevant parts of the source's API.  To me, GA means that we can both test oAuth and have a sandbox/test account that provides data for all the streams we provide.  So, I would suggest leaving the connector in beta until we get a test account with oAuth.  If I had to choose, I think ensuring correct data (e.g. `expected_records`) is more important than ensuring oAuth works (we value correctness more than anything else).

Reply:
We can split data tests and Oauth tests and set them using:
- OAuth connectivity checks and discover - prod account
- other SAT-related tests - sandbox

WDYT?

Reply:
I think that's fair! Let's do it

Reply:
@YowanR 
Sure, do we already have the sandbox so I could do it)? Or are we moving this to "on hold" until the sandbox exists?